transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -494,19 +495,27 @@ void epollInReady() {
                     do {
                         ByteBuf byteBuf = allocHandle.allocate(allocator);
                         final boolean read;
-                        if (connected) {
-                            read = connectedRead(allocHandle, byteBuf);
-                        } else {
-                            int datagramSize = config().getMaxDatagramPayloadSize();
-                            int numDatagram = datagramSize == 0 ? 1 : byteBuf.writableBytes() / datagramSize;
+                        int datagramSize = config().getMaxDatagramPayloadSize();
+                        int numDatagram = datagramSize == 0 ? 1 : byteBuf.writableBytes() / datagramSize;
 
+                        try {
                             if (numDatagram <= 1) {
-                                read = read(allocHandle, byteBuf, datagramSize);
+                                if (connected) {
+                                    read = connectedRead(allocHandle, byteBuf);
```

Comment:
I think similar changes would be needed in `connectedRead` as were made to `read` w.r.t. `datagramSize` truncation and byte counts reported to the RecvByteAllocator?

Reply:
good point ... let me address this

Reply:
done PTAL again